### ISSUE TITLE
Defaults for null values

### ISSUE THREAD

**Issue by nolledge** [circe/circe-derivation#115]
As I was trying to replace circe-generic with circe derivation in a project, I recognized that the behaviour regarding default values was different. When a null value is present the case class will not be constructed with the default value

```
@ @JsonCodec case class MyData(strings: Seq[String] = Seq.empty)
defined class MyData

@ import io.circe.parser._
import io.circe.parser._

@ decode[MyData]("""{"strings": null}""")
res14: Either[io.circe.Error, MyData] = Left(DecodingFailure(C[A], List(DownField(strings))))

@ decode[MyData]("""{}""")
res15: Either[io.circe.Error, MyData] = Right(MyData(List()))
```

I expectd MyData to be constructed with an empty string in the first case.

**Comment by nolledge** [comment]
This is my current workaround
```
    implicit val myDecoder: Decoder[MyData] = deriveDecoder.prepare(_.withFocus(_.dropNullValues))

```

**Comment by travisbrown** [comment]
Fixed in #117.
